### ISSUE TITLE
option_as_ref_deref no longer catches direct calls to Deref::deref or DerefMut::deref_mut

### ISSUE THREAD

**Issue by Xaeroxe** [rust-lang/rust-clippy#5927]
I tried this code:

```rust
if let Some(value) = value.as_ref().map(std::ops::Deref) {
  //...
}
```

I expected to see this happen: Lint option_ref_as_deref is triggered and I get redirected to `.as_deref()`

Instead, this happened: No lint was triggered

Playground for convenience: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7e8a3b2ac13f0d98a9f79a095b2fd8f2

### Meta

- `cargo clippy -V`: clippy 0.0.212 (d3fb005 2020-07-31)
- `rustc -Vv`:
  ```
  rustc 1.45.2 (d3fb005a3 2020-07-31)
  binary: rustc
  commit-hash: d3fb005a39e62501b8b0b356166e515ae24e2e54
  commit-date: 2020-07-31
  host: x86_64-unknown-linux-gnu
  release: 1.45.2
  LLVM version: 10.0
  ```

Backtrace: N/A, this bug is something **not** happening when it was expected.
